File: data/data.code.javascript.client/AckChatSendOk.js
module.exports = AckChatSendOk;

var Packet = require('../net/Packet');


function AckChatSendOk() {
	this._channel = 0;
	this._uid = 0;
	this._uname = "";
	this._content = "";


	this.Decode = function(packet) {
		this._channel = packet.ReadByte();
		this._uid = packet.ReadUint();
		this._uname = packet.ReadString();
		this._content = packet.ReadString();
	}
}

AckChatSendOk.prototype = {
	set Channel(val) {
		this._channel = val;
	},
	get Channel() {
		return this._channel;
	},

	set Uid(val) {
		this._uid = val;
	},
	get Uid() {
		return this._uid;
	},

	set Uname(val) {
		this._uname = val;
	},
	get Uname() {
		return this._uname;
	},

	set Content(val) {
		this._content = val;
	},
	get Content() {
		return this._content;
	},
}

File: examples/lua/client/protocols/AckChatSendOk.lua
module("AckChatSendOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckChatSendOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.channel = 0
	self.uid = 0
	self.uname = 0
	self.content = 0
end


function decode(self, pack)
	self.channel = pack:readU8()
	self.uid = pack:readU32()
	self.uname = pack:readString()
	self.content = pack:readString()
	return self
end
